### PR TITLE
Add unsupported flag for telemetry

### PR DESCRIPTION
**Pull request by jennchenn**
**What does this PR do?**

To protect against older agent versions that do not support telemetry, when the requested path is not found, set `@unsupported=true` in the `Client` and disable future telemetry events. Additionally, this changes all log levels related to telemetry to `debug` rather than `info`.

**Motivation**

These changes were made to prevent errors from telemetry causing unnecessary info logs from being logged.

**Additional Notes**

**How to test the change?**

Unit tests were added to test the new expected behavior

### CODE REVIEW

File: lib/datadog/core/telemetry/client.rb [hunk 1]
```diff
@@ -41,7 +44,16 @@ def disable!
         def started!
           return unless @enabled
 
-          @emitter.request('app-started')
+          res = @emitter.request('app-started')
+
+          # Telemetry is only supported by agent versions 7.34 and up
+          if res.not_found?
+            Datadog.logger.debug('Agent does not support telemetry; disabling future telemetry events.')
+            @enabled = false
```
Comment (marcotc) [inline lib/datadog/core/telemetry/client.rb:52]: You probably want to call `disable!` here, in order to stop the worker as well.

Comment (marcotc) [reply lib/datadog/core/telemetry/client.rb:52]: Actually the worker wasn't started yet 🤔

Comment (jennchenn) [reply lib/datadog/core/telemetry/client.rb:52]: The worker starts up after `started!` gets called - to ensure the `app-started` event gets sent before `heartbeat` -- would it be better to initialize the worker inside of `#started!` instead?

Comment (marcotc) [reply lib/datadog/core/telemetry/client.rb:51]: I think it's ok as is today then.

**Review by marcotc — APPROVED** [review]

**Pull request merged**